run-tests.sh

```diff
@@ -10,5 +10,5 @@ pydocstyle invenio_db && \
 isort -rc -c -df **/*.py && \
 check-manifest --ignore ".travis-*" && \
 sphinx-build -qnNW docs docs/_build/html && \
-python setup.py test && \
+python -m pytest && \
```

Comment:
Can't we just run `pytest`?
```console
$ pytest
```

Reply:
Yes, the behavior is the same if everything is installed properly. However if the user forgets to install the test requirements, then the system will look for a `pytest` executable outside of the venv. If the package happens to be installed globally then that version will be executed: the result is unpredictable and the problem can be hard to discover. This also apply to a lesser extent to the other commands (`isort`, `spinx`, etc.). Pinging @frankois

Reply:
Thanks a lot for the explanation @FlorianCassayre, this is exactly the kind of things we should spot to make developers' life easier (in the context of this sprint and in general) and it definitely helps to make CI builds reproducible. This change should be propagated to https://github.com/inveniosoftware/cookiecutter-invenio-module would you take care of it?

Reply:
Sure thing, I'll take care of that tomorrow.

Reply:
@diegodelemos There is just a small caveat that I forgot to think about: since we would now be dealing with _package names_ instead of _executable names_, the commands do no necessarily have the same name. For instance here would be the changes for this repository:
```diff
-pydocstyle invenio_db && \
-isort -rc -c -df **/*.py && \
-check-manifest --ignore ".travis-*" && \
-sphinx-build -qnNW docs docs/_build/html && \
+python -m pydocstyle invenio_db && \
+python -m isort -rc -c -df **/*.py && \
+python -m check_manifest --ignore ".travis-*" && \
+python -m sphinx.cmd.build -qnNW docs docs/_build/html && \
 python -m pytest && \
-sphinx-build -qnNW -b doctest docs docs/_build/doctest
+python -m sphinx.cmd.build -qnNW -b doctest docs docs/_build/doctest
```

While making it safer, this might confuse (new) users. Is it worth the price? Similar changes would be required for the cookie-cutter.

Reply:
I think the argument of having more reproducible tests (for which we even had a dedicated issue https://github.com/inveniosoftware/invenio/issues/4037) would be stronger than the possible confusion it may create. What if:
- We try to find another way of enforcing the commands to be inside the virtualenv (a very naive example would be to prefix the commands with the full path to the executables of the current Python installation, which should fail if the commands do not exists)
- If the previous is not possible, we document it somehow in the code with a comment to avoid confusion?

Reply:
Closing this thread as it was solved in https://github.com/inveniosoftware/pytest-invenio/issues/56#issuecomment-680726765.